src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+export type Shapes = 'triangle' | 'circle' | 'rectangle';
+export type Colors = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
 
-}
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0
+      || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('your error message');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const areaFull = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
+    return Number(areaFull.toFixed(2));
```

Comment:
you can do like that
So, create a separate function & reuse this function
![image](https://user-images.githubusercontent.com/91411692/230886181-b7a1f6d0-4aee-4284-bf7c-48ff546f37a9.png)